Tasks/JavaToolInstallerV0/chocoJavaInstall.ps1

```diff
@@ -0,0 +1,89 @@
+param (
+	[string]$VersionSpec
+    )
+
+# Set new machine path 
+function Set-MachinePath{
+    [CmdletBinding()]
+    param(
+        [string]$NewPath
+    )
+    Set-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name Path -Value $NewPath
+    return $NewPath
+}
+
+# Get machine path from registry
+function Get-MachinePath{
+    [CmdletBinding()]
+    param(
+
+    )
+    $currentPath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name PATH).Path
+    return $currentPath
+}
+
+$jdkVersion = ("jdk" + $VersionSpec)
+$jdkMatchString = ('jdk*' + $VersionSpec + '*')
+$extendedJavaHome = ('JAVA_HOME_' + $VersionSpec + '_X64')
+
+Write-Host "Installing chocolatey"
+$chocoExePath = 'C:\ProgramData\Chocolatey\bin'
```

Comment:
ProgramData may not be on the C: drive. You can use an environment variable to construct the path instead. Run `set | findstr ProgramData` to identify the environment variable. There's also the [Environment.SpecialFolder enum](https://docs.microsoft.com/en-us/dotnet/api/system.environment.specialfolder?view=netframework-4.7.2), but I didn't spot an entry that's obviously ProgramData.